Kickstarter-iOS/Features/PledgeView/Controllers/PostCampaignCheckoutViewController.swift

```diff
@@ -259,11 +251,35 @@ final class PostCampaignCheckoutViewController: UIViewController, MessageBannerV
     )
 
     let navigationController = UINavigationController(rootViewController: loginSignupViewController)
-    let navigationBarHeight = navigationController.navigationBar.bounds.height
 
     self.present(navigationController, animated: true)
   }
 
+  private func confirmPayment(with validation: PaymentSourceValidation) {
+    guard validation.requiresConfirmation else {
```

Comment:
This could probably be done in a highly clever was in the view model, but I felt that this was going to be the easiest to follow. This way, new and existing cards go through the same input/output flow, all the time.

Reply:
Yeah, I like this